src/test/java/spoon/MavenLauncherTest.java

```diff
@@ -219,4 +237,22 @@ public void testGuessMavenHome() {
 		assertTrue(mavenHome.exists());
 		assertTrue(mavenHome.isDirectory());
 	}
+
+	@Test
+	void mavenLauncherPassesEnvironmentVariables() {
+		// 10.1.0
+		MavenLauncher launcher = new MavenLauncher("./src/test/resources/maven-launcher/with-environment-variables", MavenLauncher.SOURCE_TYPE.ALL_SOURCE);
+		launcher.addEnvironmentVariable("spoonVersion", "10.1.0");
+		launcher.rebuildClasspath();
+
+		boolean containsSpoonDependency = Arrays
+			.stream(launcher.getEnvironment().getSourceClasspath())
+			.anyMatch(it -> it.matches(".*fr.inria.gforge.spoon.spoon-core.10.1.0.spoon-core-10.1.0.jar"));
```

Comment:
Are you entirely certain the classpath will _end_ with this dependency? Perhaps the test fails on Windows because it's somewhere else?

```suggestion
			.anyMatch(it -> it.matches(".*fr.inria.gforge.spoon.spoon-core.10.1.0.spoon-core-10.1.0.jar.*"));
```

Reply:
That's a good spot, but:
It is an array with one entry per dependency, so it *should* be fine. The [classpath printed in the CI](https://github.com/INRIA/spoon/runs/7711796963?check_suite_focus=true#step:10:579) is an empty array, so nothing is found. I could observe that happening locally if the maven invocation failed and did not produce any output, but the invocation request does not seem to throw an exception in such cases.

Reply:
It probably failed then. We've had trouble with Maven on Windows before. I'll have a peek once I get home then!